crates/services/txpool_v2/src/selection_algorithms/ratio_tip_gas.rs

```diff
@@ -0,0 +1,159 @@
+use std::{
+    cmp::{
+        Ordering,
+        Reverse,
+    },
+    collections::BTreeMap,
+    time::Instant,
+};
+
+use fuel_core_types::{
+    fuel_tx::TxId,
+    services::txpool::PoolTransaction,
+};
+use num_rational::Ratio;
+
+use crate::{
+    error::Error,
+    storage::Storage,
+};
+
+use super::{
+    Constraints,
+    SelectionAlgorithm,
+};
+
+pub type RatioTipGas = Ratio<u64>;
+
+/// Key used to sort transactions by tip/gas ratio.
+/// It first compares the tip/gas ratio, then the creation instant and finally the transaction id.
+#[derive(Eq, PartialEq, Clone, Copy, Debug)]
+pub struct Key {
+    ratio: RatioTipGas,
+    creation_instant: Instant,
+    tx_id: TxId,
+}
+
+impl Ord for Key {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let cmp = self.ratio.cmp(&other.ratio);
+        if cmp == Ordering::Equal {
+            let instant_cmp = other.creation_instant.cmp(&self.creation_instant);
+            if instant_cmp == Ordering::Equal {
+                self.tx_id.cmp(&other.tx_id)
```

Comment:
I'm wondering if this is attackable... Since Id is _"The unique identifier of the transaction is based on its content."_ one can theoretically play with the transaction content to get a "better ID".

But this would also require to hit the exact ratio and perfect timing, so not a big deal, after all.

Reply:
Yes exactly if you want to react to a transaction of someone you will always be (with the same ratio) less prioritised.